File: bri-kb-api/Models/TransactionModel.js
'use strict';

var sql = require('../Config/db');

var TransactionModel = function(transaction){
    this.IDList = transaction.IDList;
    this.IDPropinsi = transaction.IDPropinsi;
    this.NamaPropinsi = propinsi.NamaPropinsi;
    this.IDKontrasepsi = transaction.IDKontrasepsi;
    this.NamaKontrasepsi = transaction.NamaKontrasepsi;
    this.JumlahPemakaian = transaction.JumlahPemakaian;
};

TransactionModel.getData = function (result) {
    sql.query('select * from "LIST_PEMAKAI_KONTRASEPSI"', function (err, res) {             
        if(err) {
            console.log("error: ", err);
            result(err, null);
        }
        else{
            result(null, res.rows);
        
        }
    });   
};

TransactionModel.getDataReport = function (result) {
    sql.query('select * from "funcGetDataReport"()', function (err, res) {             
        if(err) {
            console.log("error: ", err);
            result(err, null);
        }
        else{
            result(null, res.rows);
        
        }
    });   
};

TransactionModel.insert = function (req, result) {
    var data = [
        req.idPropinsi,
        req.idKontrasepsi,
        req.jumlah
    ];
    var sqlExec = 'CALL "proctransactionsave"($1, $2 ,$3)';
    sql.query(sqlExec, data, function (err, res) {             
        if(err) {
            console.log("error: ", err);
            result(err, null);
        }
        else{
            result(null, res.rows);
        }
    });   
};


module.exports = TransactionModel;

File: kontrasepsi_app/lib/model/transactionModel.dart
class TransactionModel {
  final int idList;
  final int idPropinsi;
  final int idKontrasepsi;
  final int jumlahPemakaian;
  final String namaPropinsi;
  final String namaKontrasepsi;

  TransactionModel.fromJSON(Map<String, dynamic> jsonMap) :
    idList = jsonMap['idList'],
    idPropinsi = jsonMap['idPropinsi'],
    idKontrasepsi = jsonMap['idKontrasepsi'],
    jumlahPemakaian = jsonMap['jumlahPemakaian'],
    namaPropinsi = jsonMap['namaPropinsi'],
    namaKontrasepsi = jsonMap['namaKontrasepsi'];
}

class TransactionReportModel {
  final int rowNumber;
  final String namaPropinsi;
  final int pil;
  final int kondom;
  final int iud;
  final int jumlah;

  TransactionReportModel.fromJSON(Map<String, dynamic> jsonMap) :
    rowNumber = jsonMap['row_number'],
    namaPropinsi = jsonMap['nama_propinsi'],
    pil = int.parse(jsonMap['pil']),
    kondom = int.parse(jsonMap['kondom']),
    iud = int.parse(jsonMap['iud']),
    jumlah = int.parse(jsonMap['jumlah']);
}